src/ui/ui.js

```diff
@@ -205,4 +161,64 @@ https://raw.githubusercontent.com/fluid-project/sjrk-story-telling/master/LICENS
         }
     });
 
+    /**
+     * Fabricates a grade based on the model values passed in from the event
+     * This roundabout approach is necessary to ensure that we can have
+     * model values from the event merged successfully with the base values
+     * of the block
+     *
+     * @param {Object} modelValuesFromEvent - the model values to use
+     *
+     * @return {String} - the new grade's name
+     */
+    sjrk.storyTelling.ui.storyUi.getBlockGradeFromEventModelValues = function (modelValuesFromEvent) {
+        var gradeName = "sjrk.storyTelling.block-" + fluid.allocateGuid();
+        fluid.defaults(gradeName, {
```

Comment:
Me too! The result of this function is passed into a `gradeNames` key in the `block` subcomponents of the `blockManager`, for what it's worth. The exact details require some careful parsing of the `blockManager` component and the grade it's based on, `sjrk.dynamicViewComponentManager`.

It is a weird setup, but it's the one we figured out that lets us dynamically add viewComponents with varying grades.

Reply:
This isn't recommended since in current Infusion these defaults definitions will leak - whilst dynamically generated grades are used from time to time, this is only where there is no alternative. It seems like a better approach would be to ensure that the values fired in the event are already properly defaulted (in createBlocksFromData ?) and then to write the dynamic references to the event arguments inline in the definition of the "block" component.
General notes - 
- yes, supplying no grades at all in "mixin" grades is a reasonable technique, and was even sometimes required in the old framework where it was possible that trying to mix in "fluid.modelComponent" to a view component again would trash its argument list. This bug got fixed in the new framework.
- the definition of "storyUi" is too large and should be broken up at least at one point, perhaps at "blockManager", "managedViewComponents", or both. It would be helpful to supply at least some kind of base grade for "managedViewComponents" in any case to guide the reader as to what kind of material might get created.

Reply:
I agree with these suggestions (and thank you for them!), though I feel they're out of scope for this Jira.

I went to file a Jira to capture your feedback, but there are in fact four(!) that already address reviewing the DVCM and its usage as the blockManager, and hopefully the time to tackle them will come soon :)

The most relevant issues are [SJRK-115](https://issues.fluidproject.org/browse/SJRK-115) and [SJRK-126](https://issues.fluidproject.org/browse/SJRK-126), while [SJRK-61](https://issues.fluidproject.org/browse/SJRK-61) and [SJRK-262](https://issues.fluidproject.org/browse/SJRK-262) are a little more tangentially relevant.

Reply:
@BlueSlug if these issues will not be addressed in this work, it would be useful to track the related issues in comments with the relevant code.